react-main/src/app/screens/Game/index.js

```diff
@@ -1,17 +1,64 @@
 import React, { Component } from 'react';
 
 import Board from './components/Board';
+import { calculateWinner } from './utils';
 
 class Game extends Component {
+  state = {
+    history: [{ squares: Array(9).fill(null) }],
+    stepNumber: 0,
+    xIsNext: true
+  };
+
+  handleClick = i => {
+    const { history, xIsNext, stepNumber } = this.state;
+    const current = history.slice(0, stepNumber + 1);
+    const modifiedHistory = current[current.length - 1];
+    const squares = modifiedHistory.squares.slice();
+    if (calculateWinner(squares) || squares[i]) return;
+    squares[i] = xIsNext ? 'X' : 'O';
+    this.setState({
+      history: history.concat([{ squares }]),
+      xIsNext: !xIsNext,
+      stepNumber: history.length
+    });
+  };
+
+  jumpTo = step => {
+    this.setState({
+      stepNumber: step,
+      xIsNext: step % 2 === 0
+    });
+  };
+
   render() {
+    const { xIsNext, history, stepNumber } = this.state;
+    const current = history[stepNumber];
+    const winner = calculateWinner(current.squares);
+    const moves = history.map((step, move) => {
```

Comment:
I would extract moves to a component, that receives history from this component state, and returns its layout. That way, you wouldn't be generating this list each time the board is re-rendered